tests/components/vallox/test_switch.py

```diff
@@ -0,0 +1,62 @@
+"""Tests for Vallox switch platform."""
+import pytest
+
+from homeassistant.components.switch.const import DOMAIN as SWITCH_DOMAIN
+from homeassistant.const import ATTR_ENTITY_ID, SERVICE_TURN_OFF, SERVICE_TURN_ON
+from homeassistant.core import HomeAssistant
+
+from .conftest import patch_metrics, patch_metrics_set
+
+from tests.common import MockConfigEntry
+
+
+@pytest.mark.parametrize(
+    "entity_id, metric_key, value",
+    [
+        ("switch.vallox_bypass_locked", "A_CYC_BYPASS_LOCKED", 1),
+        ("switch.vallox_bypass_locked", "A_CYC_BYPASS_LOCKED", 0),
+    ],
+)
+async def test_switch_entities(
+    entity_id, metric_key, value, mock_entry: MockConfigEntry, hass: HomeAssistant
+):
+    """Test switch entities."""
+    # Arrange
+    metrics = {metric_key: value}
+
+    # Act
+    with patch_metrics(metrics=metrics):
+        await hass.config_entries.async_setup(mock_entry.entry_id)
+        await hass.async_block_till_done()
+
+    # Assert
+    sensor = hass.states.get(entity_id)
+    assert sensor.state == "on" if value else "off"
```

Comment:
Add a separate parameter for the expected state instead of using conditional logic.